### PR TITLE
Introduce torus template command

### PR DESCRIPTION
**Pull request by ianlivingstone**
The `torus template` command allows a user to inject secrets stored in
torus as they derive a configuration file from a template.

The current templating engine is the [raymond
handlebars](https://github.com/aymerick/raymond) engine for Go
which supports v3 of the handlebars spec.

### Example Usage

Simple variable replacement example with a json file:

```bash
$ cat test.json.template
{
    "message": "{{ message }}",
    "port": {{ port }}
}
$ ./torus template test.json.template
{
    "message": "this value was set using ag",
    "port": 34234
}
```

Outputting to a file instead of stdout using the optional output parameter:

```bash
$ ./torus template test.json.template config.json
$ cat config.json
{
    "message": "this value was set using ag",
    "port": 34234
}
```

Iterating over the special `_config` variable:

```bash
$ cat test.json.template
{
    "message": "{{ message }}",
    "port": {{ port }},
    "config": {
    {{#each _config}}
        "{{ name }}": "{{ value }}"{{#unless @last}},{{/unless}}
    {{/each}}
    }
}
$ ./torus template test.json.template
{
    "message": "this value was set using ag",
    "port": 4432,
    "config": {
        "message": "this value was set using ag",
        "port": "4432",
        "test_db": "testdburl"
    }
}
```

Closes manifoldco/torus-cli#51

### CODE REVIEW

**Comment by ianlivingstone** [comment]
This works great for the most basic use-case, however, to iterate over the exposed configuration items and inject them into a json file isn't exactly ideal:

```
$ cat test.json.template
{
    "message": "{{ message }}",
    "port": {{ port }},
    "config": {
    {{#each _config}}
        "{{ name }}": "{{ value }}"{{#unless @last}},{{/unless}}
    {{/each}}
    }
}
$ ./torus template test.json.template
{
    "message": "this value was set using ag",
    "port": 4432,
    "config": {
        "message": "this value was set using ag",
        "port": "4432",
        "test_db": "testdburl"
    }
}
```

A few problems:

- Your template has to produce valid json -- so you'll have to do that yourself
- We don't offer a good helper to do the type outputting properly
- We don't offer a helper for converting an entire slice or map to JSON inside the template itself

These will be problems despite the engine we use for templating -- as the engine will never be aware enough to offer nice solutions for deriving JSON, YAML, or <insert config here>. To make this a nice experience, we'd have to offer helpers.

I'm not super happy with the more advanced use cases, however, it seems to me that as you got more advanced you'd probably be using something like ansible or chef to derive your configuration files. At which point, you wouldn't want to shell out, you'd want a native plugin.

Thoughts?

**Comment by ianlivingstone** [comment]
> Thoughts?

For the use cases this feature supports (I have a config file for a service, I want to put secrets inside it) the feature as coded works well. The challenges I noted above, only comes into effect if you are storing complex data (which we don't support today) *or* you want to combine multiple service configs into one config file (which this feature does not support).

Therefore, for the purposes of enabling this workflow, I think we can go ahead and merge this as implemented. Then we can begin discussing how we'd want to enable those other use cases (which I'd advocate offering language libraries and plugins to things like ansible, chef or puppet).

File: promptui/select.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func (s *Select) Run() (int, string, error) {
 }
 
 func (s *Select) innerRun(starting int, top rune) (int, string, error) {
-	stdin := readline.NewCancelableStdin()
+	stdin := readline.NewCancelableStdin(os.Stdin)
```
Comment (jeffandersen) [inline promptui/select.go]: Is this due to a change in the underlying package?

Should probably be noted as a dep upgrade/fix in the changelog if we're changing more than what goes into the feature.

Comment (ianlivingstone) [inline promptui/select.go]: Yep -- the dep got upgraded.

Should we document dependency upgrades? It doesn't seem useful to anyone but those who contribute to the project.

**Comment by jeffandersen** [comment]
Values that are found in the template but not actually set in the environment come out as `""`.

Should they not be `null`?

**Comment by jeffandersen** [comment]
I guess the problem there is that they're likely wrapped in quotes in the template...

**Comment by ianlivingstone** [comment]
Yeah -- ideally we'd error but I'm not sure if the template engine we're using provides a hook for catching non-provided template variables which would be ideal.

**Comment by ianlivingstone** [comment]
Going to look for a better package before we move ahead with this change.

**Pull request closed without merging**